metadata-ingestion/src/datahub/ingestion/source/redshift/lineage.py

```diff
@@ -427,18 +445,36 @@ def _get_upstream_lineages(
 
                 # Filtering out tables which does not exist in Redshift
                 # It was deleted in the meantime or query parser did not capture well the table name
+                # Or it might be a temp table
                 if (
                     db not in all_tables
                     or schema not in all_tables[db]
                     or not any(table == t.name for t in all_tables[db][schema])
                 ):
                     logger.debug(
-                        f"{source.urn} missing table, dropping from lineage.",
+                        f"{source.urn} missing table. Adding it to temp table list for target table {target_table}.",
                     )
+                    probable_temp_tables.append(table)
                     self.report.num_lineage_tables_dropped += 1
                     continue
 
             targe_source.append(source)
+
+        if probable_temp_tables and self.config.resolve_temp_table_in_lineage:
```

Comment:
I think this is only true for the non-sql parser lineage as our SQL parser won't return tables which doesn't exits in datahub

Reply:
Also, unfortunately, I think we should store the session ID as well with the `temp_tables.` It does matter which session/transaction the temp table was created?
For example I can have multiple session where the temp table CTAS is different but the table name is always `my_awesome_temp_table`

Reply:
> I think this is only true for the non-sql parser lineage as our SQL parser won't return tables which doesn't exits in datahub

Not 100% sure - we might be able to do this. Let me do some tests

Reply:
We discussed with @hsheth2 to not care about this now and fix it later.

Reply:
so we treat `#temptable` names as normal, which also means prepending db/schema information when it's not present

see https://github.com/datahub-project/datahub/pull/9653